src/fastmcp/server/dependencies.py

```
@@ -1162,3 +1188,122 @@ async def __aenter__(self) -> ProgressLike:
+class _TokenClaim(Dependency):  # type: ignore[misc]
+    """Dependency that extracts a specific claim from the access token."""
+
+    def __init__(self, claim_name: str):
+        self.claim_name = claim_name
+
+    async def __aenter__(self) -> str:
+        token = get_access_token()
+        if token is None:
+            raise RuntimeError(
+                f"No access token available. Cannot extract claim '{self.claim_name}'."
+            )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)